src/tests/createHeadlessForm.test.js

```diff
@@ -3677,11 +3677,17 @@ describe('createHeadlessForm', () => {
           customProperties: {
```

Comment:
todo(tests): Can you write a unit tests that covers the bug we caught internally? (the nested Field called `name`). You could mount a schema with two nested fields: `name` and `type` for example. (as both names clash with the Field structure itself.

Reply:
For the test, I guess we want to test that the fields are being returned correctly and it's not throwing an error

Reply:
The scenario you have in the MR description is perfect to use as a test scenario.